packages/mobile/src/hooks/useReachabilityEffect.ts

```diff
@@ -25,19 +25,20 @@ export const useReachabilityEffect = (
 
   const handleReachabilityStateChange = useCallback(
     (nextReachabilityState: boolean) => {
-      const wasUnreachable = includeFirstRender
-        ? !prevReachability
-        : prevReachability !== undefined && !prevReachability
+      const isFirstRenderRun =
+        includeFirstRender && prevReachability === undefined
+      const wasUnreachable = prevReachability !== undefined && !prevReachability
+      const wasReachable = prevReachability !== undefined && prevReachability
 
-      const wasReachable = includeFirstRender
-        ? prevReachability
```

Comment:
this one is false when `prevReachability` is undefined, but in the `includeFirstRender` case, we want to run the onBecomeUnreachable